plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/TaskCreation.kt

```diff
@@ -127,6 +135,7 @@ internal fun <T : BaseKtLintCheckTask> createGenerateReportsTask(
 ): TaskProvider<GenerateReportsTask> = pluginHolder.target.registerTask(
     GenerateReportsTask.generateNameForSourceSets(sourceSetName, lintType)
 ) {
+    dependsOn(project.tasks.named("runKtlintFormatOverKotlinScripts"))
```

Comment:
You might not want to introduce an actual task dependency here, but rather enforce `mustRunAfter`. Please see the last sentences of https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency

> In some cases, adding a dependency on the producing task is not desired, for example when the consumer generates reports for possible multiple tasks. In this case you can introduce an ordering between the two tasks by using Task.mustRunAfter().